R/cookies_panel.R

```diff
@@ -0,0 +1,193 @@
+              Expires = c("13 months", "13 months")
+            ),
+            caption = "",
+            caption_size = "s",
+            num_col = NULL,
+            width_overwrite = c("one-quarter", "one-quarter", "one-quarter")
+          ),
+          br(),
+          div(
+            class = "govuk-grid-row",
+            div(
+              class = "govuk-grid-column-two-thirds",
+              h2(class = "govuk-heading-l", "Change your cookie settings"),
+              div(
+                class = "govuk-form-group",
+              ),
+              div(
+                class = "govuk-form-group",
+                tags$fieldset(
+                  class = "govuk-fieldset",
+                  tags$legend(
+                    class = "govuk-fieldset__legend govuk-fieldset__legend--s",
+                    "Do you want to accept analytics cookies?"
+                  ),
+                  div(
+                    class = "govuk-radios",
+                    `data-module` = "govuk-radios",
+                    div(
+                      class = "govuk-radios__item",
+                      radioButtons(NS(id, "cookies_analytics"),
+                        label = NULL,
+                        choices = list("Yes" = "yes", "No" = "no"),
+                        selected = "no",
+                        inline = TRUE
+                      )
+                    )
+                  )
+                )
+              ),
+              actionButton(NS(id, "submit_btn"), "Save cookie settings",
+                class = "govuk-button"
+              )
+            )
+          )
+        )
+      )
+    )
+  )
+}
+
+#' cookies_panel_server
+#'
+#' @description
+#' Create the server module of DfE R-Shiny cookies dashboard panel to be used
+#' alongside cookies_panel_ui().
+#'
+#' @param id ID shared with cookies_panel_ui()
+#' @param input_cookies The cookie input passed from cookies.js (should always
+#' be reactive(input$cookies))
+#' @param google_analytics_key Provide the GA 10 digit key of the form
+#' "ABCDE12345"
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' cookies_panel_server(
+#'   id = "cookies_panel",
+#'   input_cookies = reactive(input$cookies),
+#'   google_analytics_key = "ABCDE12345"
+#' )
+#' }
+cookies_panel_server <- function(
+    id,
+    input_cookies,
+    google_analytics_key = NULL) {
+  shiny::moduleServer(id, module = function(input, output, session) {
+    shiny::observeEvent(input_cookies(), {
+      if (!is.null(input_cookies())) {
+        message("Found input cookies")
+        if (!("dfe_analytics" %in% names(input_cookies()))) {
+          message("updating radio button to no")
+          updateRadioButtons(session, "cookies_analytics", selected = "no")
+        } else {
+          print(input_cookies())
+          message("Found permission cookie")
+          if (input_cookies()$dfe_analytics == "denied") {
+            message("updating radio button to no (2)")
```

Comment:
Reckon we can move these messages now as we just wanted them for bugfixing during development.